lib/pbench/agent/tool_meister.py

```diff
@@ -248,162 +348,133 @@ def wait(self):
                 f"Tool({self.name}) does not have a start process running"
             )
 
-        self.logger.info("Waiting for transient tool %s stop process", self.name)
-        # We wait for the stop process to finish first ...
-        sts = self._wait_for_process(self.stop_process)
-        if sts is None:
-            # The stop process did not terminate gracefully after 30 seconds,
-            # so we bring out the big guns ...
-            self.stop_process.kill()
-            self.logger.error(
-                "Killed un-responsive transient tool %s stop process after"
-                " 30 seconds",
-                self.name,
-            )
-        elif sts != 0 and sts != -(signal.SIGTERM):
-            self.logger.warning(
-                "Transient tool %s stop process failed with %d", self.name, sts
-            )
-        self.stop_process = None
-
-        # ... then we wait for the start process to finish, but we only wait
-        # for 30 seconds before we kill it.
-        self.logger.info("Waiting for transient tool %s start process", self.name)
-        try:
-            sts = self.start_process.wait(timeout=30)
-        except subprocess.TimeoutExpired:
-            self.start_process.kill()
-            self.logger.warning(
-                "Killed un-responsive transient tool %s start process after"
-                " 30 seconds",
-                self.name,
-            )
-        else:
-            if sts != 0 and sts != -(signal.SIGTERM):
-                self.logger.warning(
-                    "Transient tool %s start process failed with %d", self.name, sts
-                )
-        self.start_process = None
+        # First wait for the "stop" process to do it's job ...
+        self.stop_process = self._wait_for_process_w_kill(self.stop_process, "stop")
+        # ... then we wait for the start process to finish; in either case,
+        # we'll only wait a short time before killing them.
+        self.start_process = self._wait_for_process_w_kill(self.start_process, "start")
```

Comment:
Can we have the `ith` in `self._wait_for_process_with_kill()`?  (This line is already too long, so adding three more characters to make the name clear won't really hurt.)